service/worker/workerdeployment/version_workflow.go

```diff
@@ -286,68 +296,26 @@ func (d *VersionWorkflowRunner) handleSyncState(ctx workflow.Context, args *depl
 		return nil, serviceerror.NewDeadlineExceeded("Update canceled before series workflow started")
 	}
 
-	// apply changes to "current"
-	if set := args.SetCurrent; set != nil {
-		if set.LastBecameCurrentTime == nil {
-			d.VersionState.IsCurrent = false
-		} else {
-			d.VersionState.IsCurrent = true
-			d.VersionState.LastBecameCurrentTime = set.LastBecameCurrentTime
-		}
-		if err = d.updateMemo(ctx); err != nil {
-			return nil, err
-		}
-
-		// sync to task queues
-		syncReq := &deploymentspb.SyncDeploymentVersionUserDataRequest{
-			DeploymentName: d.VersionState.DeploymentName,
-			Version:        d.VersionState.Version,
-		}
-		for tqName, byType := range d.VersionState.TaskQueueFamilies {
-			for tqType, data := range byType.TaskQueues {
-				syncReq.Sync = append(syncReq.Sync, &deploymentspb.SyncDeploymentVersionUserDataRequest_SyncUserData{
-					Name: tqName,
-					Type: enumspb.TaskQueueType(tqType),
-					Data: d.dataWithTime(data),
-				})
-			}
-		}
-		activityCtx := workflow.WithActivityOptions(ctx, defaultActivityOptions)
-		var syncRes deploymentspb.SyncDeploymentVersionUserDataResponse
-		err = workflow.ExecuteActivity(activityCtx, d.a.SyncDeploymentVersionUserData, syncReq).Get(ctx, &syncRes)
-		if err != nil {
-			// TODO: if this fails, should we roll back anything?
-			return nil, err
-		}
-		if len(syncRes.TaskQueueMaxVersions) > 0 {
-			// wait for propagation
-			err = workflow.ExecuteActivity(
-				activityCtx,
-				d.a.CheckWorkerDeploymentUserDataPropagation,
-				&deploymentspb.CheckWorkerDeploymentUserDataPropagationRequest{
-					TaskQueueMaxVersions: syncRes.TaskQueueMaxVersions,
-				}).Get(ctx, nil)
-			if err != nil {
-				return nil, err
-			}
-		}
+	state := d.GetVersionState()
+	wasAcceptingNewWorkflows := state.GetCurrentSinceTime() != nil || state.GetRampingSinceTime() != nil
+	if args.GetCurrentSinceTime() != nil {
+		state.CurrentSinceTime = args.GetCurrentSinceTime()
+	}
+	if args.GetRampingSinceTime() != nil {
+		state.RampingSinceTime = args.GetRampingSinceTime()
 	}
+	isAcceptingNewWorkflows := state.GetCurrentSinceTime() != nil || state.GetRampingSinceTime() != nil
 
-	// TODO (Shivam) - Move this when UpdateMetadata is implemented.
+	if wasAcceptingNewWorkflows && !isAcceptingNewWorkflows {
+		workflow.ExecuteChildWorkflow(ctx, WorkerDeploymentCheckDrainageWorkflowType)
```

Comment:
todo: if an update causes the version to start accepting workflows again, send a cancel request to the child workflow